ports-of-call/array.hpp

```diff
@@ -0,0 +1,270 @@
+#ifndef _PORTS_OF_CALL_ARRAY_HPP_
+#define _PORTS_OF_CALL_ARRAY_HPP_
+
+#include "portability.hpp"
+#include "portable_errors.hpp"
+
+#include <cstddef>
+#include <iterator>
+#include <tuple>
+
+namespace PortsOfCall {
```

Comment:
@Yurlungur, is this namespacing correct?

Reply:
Yep :+1: